client/disputes/evidence/index.js

```diff
@@ -133,28 +134,53 @@ export const DisputeEvidenceForm = ( props ) => {
 			isLoading,
 			isDone,
 			error,
-			help: expandHelp( field.description ),
+			help: readOnly ? '' : expandHelp( field.description ),
 		};
 	};
 
 	const composeFieldControl = ( field ) => {
 		switch ( field.type ) {
 			case 'file':
-				return (
+				return ! readOnly ? (
```

Comment:
This condition is inconsistent with `readOnly && ! evidence[ field.key ]`, why so (I'm probably lacking context here)?

Reply:
All other cases are text-based fields - text, text area and date. Here if the form is submitted ( indicated by `readOnly`) , and if there is no text information submitted ( indicated by `! evidence[ field.key ]` , leading to text submitted ), we show a static text mentioning `No information submitted` . 

In case of `file` upload field though, things are slightly different. 

We are testing only for `readOnly`. If the form is `readOnly` then we show a modified file upload component - [UploadedReadOnly](https://github.com/Automattic/woocommerce-payments/blob/54cfd573047adf349e074264ca2d967e4b3ade50/client/components/file-upload/index.tsx#L139). It is similar to the original `FileUploadControl` component, with just that all sections such as - dropzone, upload button, upload error handling, are stripped out. Only the file preview, showing file name is shown . If there is no file, then a static message is shown indicating evidence file was not uploaded.  

The `UploadedReadOnly` component [has the logic](https://github.com/Automattic/woocommerce-payments/blob/54cfd573047adf349e074264ca2d967e4b3ade50/client/components/file-upload/index.tsx#L151) to detect presence or absence of an uploaded file, and show message accordingly.

Reply:
@nagpai While I agree with your reasoning for excluding the `! evidence[ field.key ]` check, I think the `readOnly` condition should be reversed here to make it consistent with the rest.

```javascript
return readOnly ? <UploadedReadOnly /> : <FileUploadControl />;
```

Reply:
Thanks for catching that, @anu-rock! I have changed[ it here](https://github.com/Automattic/woocommerce-payments/pull/4840/files#diff-6730096ed91d184adb48f33188e02b51841e49c6472b39b89b69ba2e112124cbR144).